// miui/Calendar/MonthSelect.js
var __vueify_style_dispose__ = require("vueify/lib/insert-css").insert(".month-select {\n  height: 32px;\n  max-width: 280px;\n  font-size: 18px;\n  line-height: 32px;\n  color: #333;\n}\n.month-select .float.ileft {\n  float: left;\n  font-size: 16px;\n}\n.month-select .float.iright {\n  float: right;\n  font-size: 16px;\n}\n.month-select input {\n  font-size: 18px;\n  line-height: 32px;\n  color: #333;\n  width: 60px;\n  text-align: center;\n  border: none;\n  background-color: transparent;\n}\n.month-select > i {\n  margin: 0 12px;\n  line-height: 32px;\n  cursor: pointer;\n  color: #707070;\n}\n.month-select > i:hover {\n  color: #42b983;\n}\n.month-select .disabled {\n  color: #EEE;\n}")
;(function(){
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _moment = require('moment');

var _moment2 = _interopRequireDefault(_moment);

var _ui = require('../base/ui');

var _ui2 = _interopRequireDefault(_ui);

var _MonthPicker = require('./MonthPicker');

var _MonthPicker2 = _interopRequireDefault(_MonthPicker);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var monthArray = new Array(12);

for (var idx = 0; idx < monthArray.length; idx++) {
    monthArray[idx] = idx < 9 ? '0' + (idx + 1) : '' + (idx + 1);
}

var params = {
    view: 'month',
    canPrev: true,
    canNext: true,
    year: 2016,
    month: 0,
    showMonth: 1
};

exports.default = {
    mixins: [_ui2.default],
    params: params,
    computed: {
        maxYear: function maxYear() {
            return new Date().getFullYear();
        },
        minYear: function minYear() {
            return new _moment2.default().add(-2, 'year').year();
        }
    },
    methods: {
        render: function render() {
            this.emit('month-change', {
                year: +this.year,
                showMonth: +this.showMonth,
                month: +this.month
            });
        },
        changeYear: function changeYear() {
            if (this.year < 1970 || this.year > 2500) {
                this.year = new _moment2.default(this.value).year();
            }
            this.render();
        },
        changeMonth: function changeMonth() {
            if (this.showMonth < 1 || this.showMonth > 12) {
                this.showMonth = new _moment2.default(this.value).month() + 1;
            }
            this.month = this.showMonth - 1;
            this.render();
        },
        prev: function prev(e) {
            e.stopPropagation();
            var time = new _moment2.default(new Date(this.year, this.month, 1)).subtract(1, 'month');
            this.year = time.year();
            this.month = time.month();
            this.render();
        },
        next: function next(e) {
            e.stopPropagation();
            var time = new _moment2.default(new Date(this.year, this.month, 1)).add(1, 'month');
            this.year = time.year();
            this.month = time.month();
            this.render();
        },
        showMonthPicker: function showMonthPicker() {
            this.view = 'month';
        },
        selectMonth: function selectMonth(e) {
            this.month = e.data.value - 1;
            this.showMonth = e.data.value;
            this.view = 'dates';
            this.render();
        }
    },
    components: {
        MonthPicker: _MonthPicker2.default
    }
};
})()
if (module.exports.__esModule) module.exports = module.exports.default
var __vue__options__ = (typeof module.exports === "function"? module.exports.options: module.exports)
if (__vue__options__.functional) {console.error("[vueify] functional components are not supported and should be defined in plain js files using render functions.")}
__vue__options__.render = function render () {var _vm=this;var _h=_vm.$createElement;var _c=_vm._self._c||_h;return _c('div',{staticClass:"month-select"},[_c('i',{staticClass:"sm-arrow-left float ileft",class:{disabled: !_vm.canPrev},attrs:{"disabled":!_vm.canPrev},on:{"click":_vm.prev,"touchstart":_vm.prev}}),_vm._v(" "),_c('i',{staticClass:"sm-arrow-right float iright",class:{disabled: !_vm.canNext},attrs:{"disabled":!_vm.canNext},on:{"click":_vm.next,"touchstart":_vm.next}}),_vm._v(" "),_c('div',{staticClass:"text center"},[_c('input',{directives:[{name:"model",rawName:"v-model",value:(_vm.year),expression:"year"}],staticClass:"year-input",attrs:{"type":"number","min":_vm.minYear,"max":_vm.maxYear},domProps:{"value":_vm.year,"value":_vm._s(_vm.year)},on:{"change":function($event){_vm.changeYear()},"input":function($event){if($event.target.composing){ return; }_vm.year=_vm._n($event.target.value)},"blur":function($event){_vm.$forceUpdate()}}}),_vm._v("\n         / \n        "),_c('input',{directives:[{name:"model",rawName:"v-model",value:(_vm.showMonth),expression:"showMonth"}],staticClass:"month-input",attrs:{"type":"number","min":"1","max":"12"},domProps:{"value":_vm.showMonth,"value":_vm._s(_vm.showMonth)},on:{"change":function($event){_vm.changeMonth()},"click":_vm.showMonthPicker,"input":function($event){if($event.target.composing){ return; }_vm.showMonth=_vm._n($event.target.value)},"blur":function($event){_vm.$forceUpdate()}}})]),_vm._v(" "),(_vm.view=='month')?_c('month-picker',{attrs:{"params":{value: _vm.showMonth}},on:{"value-change":_vm.selectMonth}}):_vm._e()],1)}
__vue__options__.staticRenderFns = []
if (module.hot) {(function () {  var hotAPI = require("vueify/node_modules/vue-hot-reload-api")
  hotAPI.install(require("vue"), true)
  if (!hotAPI.compatible) return
  module.hot.accept()
  module.hot.dispose(__vueify_style_dispose__)
  if (!module.hot.data) {
    hotAPI.createRecord("data-v-439b69f3", __vue__options__)
  } else {
    hotAPI.reload("data-v-439b69f3", __vue__options__)
  }
})()}
