// miui/LineBox/LineBox-aa.js
var __vueify_style_dispose__ = require("vueify/lib/insert-css").insert(".line-box {\n  position: relative;\n  border: 1px solid #e4e4e4;\n  border-radius: 3px;\n  min-height: 10rem;\n  box-sizing: content-box;\n  overflow: hidden;\n}\n.line-box .line-list {\n  display: block;\n  width: 30px;\n  text-align: center;\n  background-color: #e3e3e3;\n  padding-top: 2px;\n  font-size: 13px;\n  height: 100%;\n  overflow: hidden;\n  position: absolute;\n  top: 0;\n  left: 0;\n}\n.line-box .line-list li {\n  height: 25px;\n  line-height: 25px;\n  text-align: center;\n}\n.line-box .textarea-container {\n  padding-left: 35px;\n  box-sizing: border-box;\n}\n.line-box textarea {\n  width: 100%;\n  font-size: 13px;\n  border: 0;\n  outline: none;\n  line-height: 25px;\n  min-height: 200px;\n  overflow: hidden;\n  padding-left: 5px;\n}")
;(function(){
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _ui = require('../base/ui');

var _ui2 = _interopRequireDefault(_ui);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var params = {
    content: '',
    rows: 10,
    cols: 30,
    doms: {
        textarea: null,
        ulNum: null
    },
    textareaDisabled: false
};

exports.default = {
    params: params,
    mixins: [_ui2.default],
    computed: {
        style: function style() {
            return {
                width: this.cols + 'rem',
                height: this.rows + 'rem'
            };
        }
    },
    methods: {
        textChanged: function textChanged(e) {
            var lines = this.content.split('\n');
            this.rows = lines.length;
            this.emit('content-change', {
                content: this.content,
                lines: lines
            });
        }
    },
    watch: {
        content: function content() {
            this.textChanged();
        }
    }
};
})()
if (module.exports.__esModule) module.exports = module.exports.default
var __vue__options__ = (typeof module.exports === "function"? module.exports.options: module.exports)
if (__vue__options__.functional) {console.error("[vueify] functional components are not supported and should be defined in plain js files using render functions.")}
__vue__options__.render = function render () {var _vm=this;var _h=_vm.$createElement;var _c=_vm._self._c||_h;return _c('div',{staticClass:"line-box",style:(_vm.style)},[_c('ul',{staticClass:"line-list"},_vm._l((_vm.rows),function(row){return (_vm.rows > 0)?_c('li',[_vm._v(_vm._s(row))]):_vm._e()})),_vm._v(" "),_c('div',{staticClass:"textarea-container",style:(_vm.style)},[_c('textarea',{directives:[{name:"model",rawName:"v-model",value:(_vm.content),expression:"content"}],staticClass:"input-textarea",attrs:{"disabled":_vm.textareaDisabled == 1},domProps:{"value":_vm._s(_vm.content)},on:{"input":function($event){if($event.target.composing){ return; }_vm.content=$event.target.value}}})])])}
__vue__options__.staticRenderFns = []
if (module.hot) {(function () {  var hotAPI = require("vueify/node_modules/vue-hot-reload-api")
  hotAPI.install(require("vue"), true)
  if (!hotAPI.compatible) return
  module.hot.accept()
  module.hot.dispose(__vueify_style_dispose__)
  if (!module.hot.data) {
    hotAPI.createRecord("data-v-176feeb6", __vue__options__)
  } else {
    hotAPI.reload("data-v-176feeb6", __vue__options__)
  }
})()}
