// miui/LineBox/LineBox.js
var __vueify_style_dispose__ = require("vueify/lib/insert-css").insert(".text-area-lines {\n  position: relative;\n  border: 1px solid #e4e4e4;\n  border-radius: 3px;\n  height: 302px;\n}\n.text-area-lines .error-line {\n  color: red;\n}\n.text-area-lines .error-info {\n  position: absolute;\n  top: -10000px;\n  left: 5px;\n}\n.text-area-lines .error-info .sm-question {\n  width: 14px;\n  height: 12px;\n  display: inline-block;\n  cursor: pointer;\n  border: 1px solid red;\n  margin: 4px;\n  opacity: 0;\n}\n.text-area-lines .line-num {\n  display: block;\n  width: 30px;\n  text-align: center;\n  background-color: #e3e3e3;\n  float: left;\n  padding-top: 2px;\n  font-size: 13px;\n  overflow: hidden;\n  color: #666;\n}\n.text-area-lines .line-num .sm-question {\n  cursor: pointer;\n}\n.text-area-lines .line-num li {\n  height: 28px;\n  line-height: 28px;\n  text-align: right;\n  padding-right: 8px;\n}\n.text-area-lines .input-textarea-container {\n  position: relative;\n  float: left;\n}\n.text-area-lines .input-textarea[disabled] {\n  cursor: not-allowed;\n  background-color: #eee;\n}\n.text-area-lines .input-textarea {\n  width: 100%;\n  font-size: 13px;\n  border: 0;\n  outline: none;\n  line-height: 28px;\n  overflow: auto;\n  padding-left: 8px;\n  padding-top: 2px;\n  resize: none;\n}")
;(function(){
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _ui = require('../base/ui');

var _ui2 = _interopRequireDefault(_ui);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var params = {
    defaultText: '',
    height: 300,
    width: 500,
    textareaDisabled: 0
};

exports.default = {
    params: params,
    mixins: [_ui2.default],
    ready: function ready() {
        this.textareaText = this.defaultText || '';
    },

    computed: {
        containerStyle: function containerStyle() {
            return {
                height: this.height + 'px',
                width: this.width + 'px'
            };
        },
        lineStyle: function lineStyle() {
            return {
                width: '38px',
                height: this.height + 'px'
            };
        },
        txtStyle: function txtStyle() {
            return {
                height: this.height - 2 + 'px',
                width: this.width - 40 + 'px'
            };
        },
        textareaArr: function textareaArr() {
            return this.textareaText.split('\n');
        }
    },
    methods: {
        getDoms: function getDoms() {
            var cDoms = this.doms;
            if (!cDoms.textarea || !cDoms.ulNum) {
                var $pDoms = this.$el;
                cDoms.textarea = $pDoms.querySelector('.input-textarea');
                cDoms.ulNum = $pDoms.querySelector('.line-num');
                cDoms.tipDom = $pDoms.querySelector('.error-info');
            }
            return cDoms;
        },

        mouseInTextline: function mouseInTextline(e, index) {
            if (this.tipSysObj.tipFlag) {
                var errmsg = this.tipSysObj.errorObj[this.textareaArr[index]];
                var $doms = this.getDoms();
                if (errmsg) {
                    var tipTop = index * 22 - $doms.ulNum.scrollTop;
                    this.tipSysObj.curErrIndex = index;
                    this.tipSysObj.errorTip = errmsg;
                    $doms.tipDom.style.top = tipTop + 'px';
                } else if (this.tipSysObj.curErrIndex == index) {
                    this.tipSysObj.curErrIndex = -1;
                    $doms.tipDom.style.top = -10000 + 'px';
                }
            }
        },
        textareaFocus: function textareaFocus(e) {
            this.textareaText == '' && (this.lineCount = 1);
        },
        textareaBlur: function textareaBlur(e) {
            this.textareaText == '' && (this.lineCount = 0);
        },
        textareaScroll: function textareaScroll() {
            var $doms = this.getDoms();
            var cDom = $doms.textarea;
            var ulDom = $doms.ulNum;
            ulDom.scrollTop = cDom.scrollTop;
            ulDom.style.height = cDom.clientHeight + 'px';
        },

        setErrorObj: function setErrorObj(obj) {
            this.tipSysObj.tipFlag = true;
            Vue.set(this.tipSysObj, 'errorObj', obj);
            var me = this;
            setTimeout(function () {
                me.textareaText += '\n';
            }, 50);
        },
        textChanged: function textChanged(e) {
            var inputText = this.textareaText;
            this.lineCount = inputText.split("\n").length;
            this.textareaScroll();
            this.$emit('text-changed', inputText);
        }
    },
    watch: {
        'textareaText': function textareaText() {
            this.textChanged();
        }
    },
    data: function data() {
        var data = {
            textareaText: '',
            lineCount: 0,
            doingScrollAction: false,
            doms: {
                textarea: null,
                ulNum: null
            },
            tipSysObj: {
                errorTip: '',
                tipFlag: false,
                curErrIndex: -1,
                errorObj: {}
            }
        };

        if (this.defaultText) {
            data.textareaText = this.defaultText;
        }
        return data;
    }
};
})()
if (module.exports.__esModule) module.exports = module.exports.default
var __vue__options__ = (typeof module.exports === "function"? module.exports.options: module.exports)
if (__vue__options__.functional) {console.error("[vueify] functional components are not supported and should be defined in plain js files using render functions.")}
__vue__options__.render = function render () {var _vm=this;var _h=_vm.$createElement;var _c=_vm._self._c||_h;return _c('div',{staticClass:"text-area-lines",style:(_vm.containerStyle)},[_c('div',{staticClass:"error-info",staticStyle:{"diplay":"none"}},[_c('div',[_vm._v(_vm._s(_vm.tipSysObj.errorTip))])]),_vm._v(" "),_c('ul',{staticClass:"line-num",style:(_vm.lineStyle)},_vm._l((_vm.lineCount),function(n){return (_vm.lineCount > 0)?_c('li',{class:{'error-line': _vm.tipSysObj.errorObj[_vm.textareaArr[n]]},on:{"mouseenter":function($event){_vm.mouseInTextline($event, n)}}},[(_vm.tipSysObj.errorObj[_vm.textareaArr[n]])?_c('span',{staticClass:"sm-question"}):_c('span',[_vm._v(_vm._s(n))])]):_vm._e()})),_vm._v(" "),_c('div',{staticClass:"input-textarea-container"},[_c('textarea',{directives:[{name:"model",rawName:"v-model",value:(_vm.textareaText),expression:"textareaText"}],staticClass:"input-textarea",style:(_vm.txtStyle),attrs:{"disabled":_vm.textareaDisabled == 1,"wrap":"off"},domProps:{"value":_vm._s(_vm.textareaText)},on:{"scroll":_vm.textareaScroll,"blur":_vm.textareaBlur,"focus":_vm.textareaFocus,"input":function($event){if($event.target.composing){ return; }_vm.textareaText=$event.target.value}}})])])}
__vue__options__.staticRenderFns = []
if (module.hot) {(function () {  var hotAPI = require("vueify/node_modules/vue-hot-reload-api")
  hotAPI.install(require("vue"), true)
  if (!hotAPI.compatible) return
  module.hot.accept()
  module.hot.dispose(__vueify_style_dispose__)
  if (!module.hot.data) {
    hotAPI.createRecord("data-v-54d82ea7", __vue__options__)
  } else {
    hotAPI.reload("data-v-54d82ea7", __vue__options__)
  }
})()}
