// miui/RichTable/doc.js
'use strict';

exports.info = {
    name: 'RichTable',
    desc: '表格组件-支持选择，表头插件，表格插件'
};

exports.events = [{
    name: 'sort-field',
    data: '表头排序',
    desc: '{field, asc}'
}, {
    name: 'selected-changed',
    data: '选中的下标数组',
    desc: 'e.data'
}, {
    name: 'selected-list-changed',
    data: '选中的元素数组',
    desc: 'e.data'
}];

exports.methods = {
    getSelected: {
        args: 'begin, end',
        desc: '设置取值'
    },
    setSelected: {
        args: '',
        desc: '确定选择'
    }
};

exports.params = {
    cellTextMap: {
        default: '{}',
        type: ['Object<String, Function>'],
        desc: '单元格内容'
    },
    headTextMap: {
        default: '{}',
        type: ['Object<String, Function>'],
        desc: '表格头内容'
    },
    headPlugin: {
        default: '{}',
        type: ['Object<String, Function>'],
        desc: '表头插件'
    },
    cellPlugin: {
        default: '{}',
        type: ['Object<String, Function>'],
        desc: '单元格插件'
    },
    sortState: {
        default: '{}',
        type: ['Object<String, String>'],
        desc: '排序状态'
    },
    singleFieldSort: {
        default: 'true',
        type: ['Boolean'],
        desc: '是否单列排序'
    },
    cellComp: {
        default: '{}',
        type: ['Object<String, String>'],
        desc: '列与组件映射关系，组件需components注册/全局组件'
    },
    fieldInfo: {
        default: '{}',
        type: ['Object<String, String>'],
        desc: '表头信息插件'
    },
    selected: {
        default: '[]',
        type: ['Array'],
        desc: '选择行下标'
    },
    fields: {
        default: '[]',
        type: ['Array'],
        desc: '需要展现的列名称'
    },
    items: {
        default: '[]',
        type: ['Array'],
        desc: '表格列表数据'
    },
    scrollX: {
        default: 'false',
        type: ['Boolean'],
        desc: '是否表格内横向滚动(顶部滚动)'
    },
    scrollY: {
        default: 'false',
        type: ['Boolean'],
        desc: '是否表格内纵向滚动'
    },
    fieldWidth: {
        default: '[]',
        type: ['Object'],
        desc: '列宽度配置:{field: "180px"}'
    },
    selectable: {
        default: 'true',
        type: ['Boolean'],
        desc: '是否可以行选择'
    },
    needCheckAllPages: {
        default: '[]',
        type: ['Array'],
        desc: '是否支持跨页全选'
    },
    headRow: {
        default: '["origin"]',
        type: ['Array'],
        desc: '支持多个表头配置'
    }
};
